sys/riscv/riscv/pmap.c

```diff
@@ -2685,6 +2708,34 @@ pmap_promote_l2(pmap_t pmap, pd_entry_t *l2, vm_offset_t va,
 }
 #endif
 
+#if __has_feature(capabilities)
+static __inline uint64_t
+cheri_pte_cw(vm_page_t m, vm_prot_t prot, u_int flags)
+{
+	vm_page_astate_t mas = vm_page_astate_load(m);
+
+	// XXX Probably not here?
+	KASSERT((mas.flags & PGA_CAPDIRTY) == 0 ||
+	    (mas.flags & PGA_CAPSTORE) != 0,
+	    ("pmap inserting CAPDIRTY w/o CAPSTORE m=%p", m));
+
+	KASSERT((prot & VM_PROT_WRITE_CAP) == 0 ||
+	    (prot & VM_PROT_WRITE) != 0,
+	    ("pmap inserting PROT_WRITE_CAP w/o PROT_WRITE m=%p", m));
+
+	if ((prot & VM_PROT_WRITE_CAP) != 0) {
+		/* XXX PTE_CD from PGA_CAPDIRTY?  Not required... */
```

Comment:
Hmmm, do we want to proactively set `PTE_CD` for `(ftype & VM_PROT_WRITE_CAP) != 0` as we do for `PTE_D` and `VM_PROT_WRITE` faults?

Reply:
Good plan; done.